src/app/components/AccountInfo/DepositModal.tsx

```diff
@@ -7,20 +7,22 @@ import Copy from 'components/Copy';
 import { getDepositAddress } from 'modules/account/actions';
 import { AppState } from 'store/reducers';
 import './DepositModal.less';
+import { blockchainDisplayName, CHAIN_TYPE } from 'utils/constants';
+import { getNodeChain } from 'modules/node/selectors';
 
 interface StateProps {
   depositAddress: AppState['account']['depositAddress'];
   isFetchingDepositAddress: AppState['account']['isFetchingDepositAddress'];
   fetchDepositAddressError: AppState['account']['fetchDepositAddressError'];
   hasPassword: boolean;
-  nodeInfo: AppState['node']['nodeInfo'];
 }
 
 interface DispatchProps {
   getDepositAddress: typeof getDepositAddress;
 }
 
 interface OwnProps {
+  chain: CHAIN_TYPE;
```

Comment:
This is actually a `StateProps` prop, since it comes from the redux store. It's also usually good to type selectors as `ReturnType<typeof selectorFnName>;` to make refactors easier, if you change the selector.

Reply:
It looks like the `getNodeChain` function return type is `null || string` but you want it to be `CHAIN_TYPE`. You can fix this by updating the selector to be:
```
export const getNodeChain = (s: AppState) => ((s.node.nodeInfo && s.node.nodeInfo.chains[0]) || 'bitcoin') as CHAIN_TYPE;
```

Would it make more sense to lookup the values of `blockchainDisplayName`, `blockchainLogos`, `denominationSymbols` and `denominationNames` immediately after the `nodeInfo` is fetched from the API, doing it only once. You could do these lookups in the `handleGetNodeInfo` saga and pass it along with the `GET_NODE_INFO_SUCCESS` action, then store these values in the node redux state. 

Now in the UI components, you are just pulling the same string values from state instead of doing the lookups every time the components are rendered. And you no longer need to import `CHAIN_TYPE` and `getNodeChain` into many components to do the lookups.

Reply:
Yep, @jamaljsr has it. Typescript is seeing `'bitcoin'` here and thinking that it could be just any ol' string. The type might have been correctly inferred by changing `nodeInfo`'s type to be an array of CHAIN_TYPEs instead of strings, and to use `CHAIN_TYPE.BITCOIN` instead of `'bitcoin'`.

As for the redux store stuff, I'm fine with it as-is since I like keeping redux state light, but if there end up being a ton of UI changes as a result of chain type, it might be wise to bundle them all together into something like what @jamaljsr described.

Reply:
@wbobeirne has a good point about keeping the redux state light. An alternative approach, which is a middle ground, is to change your selector to return an object with the lookup values instead of just the chain. It still reduces the duplication of lookup code in your components, but doesn't pollute the redux state. 

Example:
```
export interface ChainInfo {
  displayName: string;
  denominationSymbols: DenominationMap;
  denominationNames: DenominationMap;
  blockchainLogo: React.ComponentType<CustomIconComponentProps>;
}

export const getChainInfo = (s: AppState): ChainInfo => {
  const chain = ((s.node.nodeInfo && s.node.nodeInfo.chains[0]) || 'bitcoin') as CHAIN_TYPE;
  return {
    displayName: blockchainDisplayName[chain],
    denominationSymbols: denominationSymbols[chain],
    denominationNames: denominationNames[chain],
    blockchainLogo: blockchainLogos[chain],
  };
}
```
Component:
```
import { ChainInfo, getChainInfo } from 'modules/node/selectors';
...
interface StateProps {
  chainInfo: ChainInfo
}
...
<Modal
  title={chainInfo.displayName + " Address"}
>

<Icon component={chainInfo.blockchainLogo} />
...
export default connect<StateProps, DispatchProps, OwnProps, AppState>(
  state => ({
    chainInfo: getChainInfo(state),
  }),
``` 
Just throwing this out there as food for the conversation, I don't want to over-complicate it. 😄